presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -243,8 +243,10 @@ public static double stLength(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
     @SqlType(DOUBLE)
     public static double stXMax(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
     {
-        OGCGeometry geometry = deserialize(input);
-        Envelope envelope = getEnvelope(geometry);
+        Envelope envelope = deserializeEnvelope(input);
+        if (envelope == null) {
+            return 0.0;
```

Comment:
It should be `null`. Can you add tests for that?

Reply:
setting the return value to `null` will not work because of the function return type of `double`. However, if I change the function return type to `Double`, it will work. I wonder if this change may affect many other parts of the code!

Reply:
I figured I should do: `return Double.NaN;` which works. I can have a test for that written as well. Does it seem reasonable?

Reply:
Please change it to `Double` and return `null`. Please also verify that there is coverage for this function call with `null` arguments

Reply:
After this change, there will be no usages of the getEnvelope function left. Could you remove that function? 

To test edge cases in these functions, you may use 'POLYGON EMPTY' shape. E.g.

assertFunction("ST_XMin(ST_GeometryFromText('POLYGON EMPTY'))", DOUBLE, null);

Also, this change fixes the NullPointerException reported in #10027. Would you add a test case and append "Fixes #10027" to the PR description?